engine/Examples/engine.log

```diff
@@ -0,0 +1,147 @@
+2023-08-25 13:33:24,958 - DEBUG	- Processing statement=(T v (P ^ Q))
+2023-08-25 13:33:24,958 - INFO	- Before proceeding, checks for unequal brackets and illegal characters are done.
+2023-08-25 13:33:24,958 - DEBUG	- Calling check_left_and_right_brackets((T v (P ^ Q)))...
+2023-08-25 13:33:24,958 - WARNING	- The number of left and right brackets are equal, returning.
+2023-08-25 13:33:24,958 - DEBUG	- Calling check_for_illegal_characters((T v (P ^ Q)))
+2023-08-25 13:33:24,958 - DEBUG	- ******************** starting input validation *******************
+2023-08-25 13:33:24,958 - DEBUG	- Calling validate_input(T)...
+2023-08-25 13:33:24,958 - DEBUG	- Checked against a regex. result(is it valid?)=True. Returning.
+2023-08-25 13:33:24,958 - DEBUG	- Calling validate_input(v)...
+2023-08-25 13:33:24,958 - DEBUG	- Checked against a regex. result(is it valid?)=True. Returning.
+2023-08-25 13:33:24,958 - DEBUG	- Calling validate_input(P)...
+2023-08-25 13:33:24,958 - DEBUG	- Checked against a regex. result(is it valid?)=True. Returning.
+2023-08-25 13:33:24,959 - DEBUG	- Calling validate_input(^)...
+2023-08-25 13:33:24,959 - DEBUG	- Checked against a regex. result(is it valid?)=True. Returning.
+2023-08-25 13:33:24,959 - DEBUG	- Calling validate_input(Q)...
+2023-08-25 13:33:24,959 - DEBUG	- Checked against a regex. result(is it valid?)=True. Returning.
+2023-08-25 13:33:24,959 - DEBUG	- ******************** finished input validation *******************
+2023-08-25 13:33:24,959 - WARNING	- There were no illegal characters found, returning.
+2023-08-25 13:33:24,959 - WARNING	- All checks are valid, proceeding with extracting all needed information.
+2023-08-25 13:33:24,959 - INFO	- Processing statement...
+2023-08-25 13:33:24,959 - DEBUG	- statement=(T v (P ^ Q))
+2023-08-25 13:33:24,960 - INFO	- Starting to analize statement...
+2023-08-25 13:33:24,960 - DEBUG	- Calling clean_whitespaces((T v (P ^ Q)))...
+2023-08-25 13:33:24,960 - DEBUG	- Returning without whitespaces: statement=(Tv(P^Q))
+2023-08-25 13:33:24,960 - INFO	- Finding variables in statement...
+2023-08-25 13:33:24,960 - INFO	- All whitespaces need to be removed, proceeding.
+2023-08-25 13:33:24,960 - DEBUG	- Calling clean_whitespaces((Tv(P^Q)))...
+2023-08-25 13:33:24,960 - DEBUG	- Returning without whitespaces: statement=(Tv(P^Q))
+2023-08-25 13:33:24,960 - WARNING	- Removed any whitespaces found in statement. Proceeding with finding variables.
+2023-08-25 13:33:24,960 - DEBUG	- statement=(Tv(P^Q))
+2023-08-25 13:33:24,960 - DEBUG	- variables_array=['T', 'P', 'Q']
+2023-08-25 13:33:24,960 - INFO	- Found all variables in statement. 
+2023-08-25 13:33:24,960 - DEBUG	- variables_arr=['T', 'P', 'Q']
+2023-08-25 13:33:24,960 - INFO	- Creating an array from a string...
+2023-08-25 13:33:24,960 - DEBUG	- Calling create_array_of_elements((Tv(P^Q)))
+2023-08-25 13:33:24,960 - DEBUG	- array_of_elements=['(', 'T', 'v', '(', 'P', '^', 'Q', ')', ')']
```

Comment:
I'd like to see more of this in info level

Reply:
Ok. Will get it done.